File: 898-transpose-matrix/transpose-matrix.ts
function transpose(matrix: number[][]): number[][] {
    const tMatrix = [];

    matrix.forEach((row, ri)=>{
        row.forEach((col, ci)=>{
            if(!tMatrix[ci]) tMatrix[ci] = []
            tMatrix[ci].push(col)
        })
    }) 
    return tMatrix;
};

File: 898-transpose-matrix/transpose-matrix.go
func transpose(matrix [][]int) [][]int {
    m := len(matrix)
    n := len(matrix[0])
    tMatrix := make([][]int, n)

    for i := range matrix {
        for j := range matrix[i] {
            if tMatrix[j] == nil {
                tMatrix[j] = make([]int, m)
            }
            tMatrix[j][i] = matrix[i][j]
        }
    }

    return tMatrix
}